File: app/src/main/java/com/zuk0/gaijinsmash/riderz/ui/activity/main/fragment/station_info/StationInfoViewModel.java
package com.zuk0.gaijinsmash.riderz.ui.activity.main.fragment.station_info;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;
import android.content.Context;
import android.os.Bundle;

import com.zuk0.gaijinsmash.riderz.data.local.entity.station_response.Station;
import com.zuk0.gaijinsmash.riderz.data.local.entity.station_response.StationXmlResponse;
import com.zuk0.gaijinsmash.riderz.data.remote.repository.StationRepository;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class StationInfoViewModel extends ViewModel {

    private StationRepository mRepository;
    private LiveData<StationXmlResponse> mStationLiveData;

    public String mStationAbbr;
    public Station mStationObject;
    public Bundle mBundle;

    @Inject
    StationInfoViewModel(StationRepository repository) {
        mRepository = repository;
        initData();
    }

    private synchronized void initData() {
        //todo: initialization logic here
    }

    public LiveData<StationXmlResponse> getStationLiveData(String abbr) {
        if(mStationLiveData == null) {
            mStationLiveData = mRepository.getStation(abbr);
        }
        return mStationLiveData;
    }

    enum StationInfo {
        Name("StationName"), Lat("StationLat"), Long("StationLong");

        private String value;

        StationInfo(String value) {
            this.value = value;
        }

        private String getValue() {
            return value;
        }
    }

    public Bundle getBundle(Station station) {
        Bundle bundle = new Bundle();
        bundle.putString(StationInfo.Name.getValue(), station.getName());
        bundle.putString(StationInfo.Lat.getValue(), String.valueOf(station.getLatitude()));
        bundle.putString(StationInfo.Long.getValue(), String.valueOf(station.getLongitude()));
        return bundle;
    }
}

File: app/src/main/java/com/zuk0/gaijinsmash/riderz/ui/activity/main/fragment/station_info/StationInfoViewModel.kt
package com.zuk0.gaijinsmash.riderz.ui.activity.main.fragment.station_info

import android.os.Bundle
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.zuk0.gaijinsmash.riderz.data.local.entity.station_response.Station
import com.zuk0.gaijinsmash.riderz.data.local.entity.station_response.StationXmlResponse
import com.zuk0.gaijinsmash.riderz.data.remote.repository.StationRepository
import javax.inject.Inject

class StationInfoViewModel
@Inject constructor(val mRepository: StationRepository) : ViewModel() {

    private var mStationLiveData: LiveData<StationXmlResponse>? = null
    var mStationAbbr: String? = null
    var mStationObject: Station? = null
    var mBundle: Bundle? = null

    @Synchronized
    private fun initData() { //todo: initialization logic here
    }

    fun getStationLiveData(abbr: String): LiveData<StationXmlResponse>? {
        if (mStationLiveData == null) {
            mStationLiveData = mRepository.getStation(abbr)
        }
        return mStationLiveData
    }

    internal enum class StationInfo(val value: String) {
        Name("StationName"), Lat("StationLat"), Long("StationLong");

    }

    fun getBundle(station: Station): Bundle {
        val bundle = Bundle()
        bundle.putString(StationInfo.Name.value, station.name)
        bundle.putString(StationInfo.Lat.value, java.lang.String.valueOf(station.latitude))
        bundle.putString(StationInfo.Long.value, java.lang.String.valueOf(station.longitude))
        return bundle
    }

    init {
        initData()
    }
}